app/helpers/samples_helper.rb

```diff
@@ -100,38 +99,61 @@ def samples_output_data(samples)
     final_result
   end
 
+  def filter_samples(samples, query)
+    if query == 'WAITING'
+      samples = samples.joins("LEFT OUTER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where("samples.status = ?  or pipeline_runs.job_status is NULL", 'created')
+    elsif query == 'FAILED'
+      db_query = [query, '1.Host Filtering-FAILED', '2.GSNAPL/RAPSEARCH alignment-FAILED', '3.Post Processing-FAILED', 'ERROR']
+      samples = samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status IN (?)", db_query)
+    elsif query == 'UPLOADING'
+      db_query = [query, '1.Host Filtering-STARTED', '1.Host Filtering-CHECKED', '1.Host Filtering-LOADED',
+                  '2.GSNAPL/RAPSEARCH alignment-STARTED', '2.GSNAPL/RAPSEARCH alignment-CHECKED', '2.GSNAPL/RAPSEARCH alignment-LOADED', '3.Post Processing-STARTED', '3.Post Processing-CHECKED']
+      samples = samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status IN (?)", db_query)
+    elsif query == 'CHECKED'
+      db_query = [query, '3.Post Processing-CHECKED']
+      samples = samples.joins("INNER JOIN pipeline_runs ON pipeline_runs.sample_id = samples.id").where(status: 'checked').where("pipeline_runs.id in (select max(id) from pipeline_runs group by sample_id)").where("pipeline_runs.job_status IN (?)", db_query)
+    else samples = samples
+    end
+    samples
+  end
+
   def samples_pipeline_run_info(samples)
     pipeline_run_info = []
     samples.each do |output|
       pipeline_run_entry = {}
-      pipeline_run_status = output.pipeline_runs.first ? output.pipeline_runs.first.job_status : nil
-      pipeline_run_entry[:job_status_description] =
-        if %w[CHECKED SUCCEEDED].include?(pipeline_run_status)
-          'COMPLETE'
-        elsif %w[FAILED ERROR].include?(pipeline_run_status)
-          'FAILED'
-        elsif %w[RUNNING LOADED].include?(pipeline_run_status)
-          'IN PROGRESS'
-        elsif pipeline_run_status == 'RUNNABLE'
-          'INITIALIZING'
+      if output.pipeline_runs.first
+        recent_pipeline_run = output.pipeline_runs.first
+        pipeline_run_entry[:job_status_description] = 'WAITING' if recent_pipeline_run.job_status.nil?
+        if recent_pipeline_run.pipeline_run_stages.present?
+          run_stages = recent_pipeline_run.pipeline_run_stages || []
+          host_filtering = run_stages ? run_stages.first.job_status : nil
```

Comment:
this is not true. It's possible we skip the host filtering for same samples. 
The right way should be 

run_stages.each do |rs|
  pipeline_run_entry[rs[:name]] = rs
end